port to new azure SDK

krnowak opened the pull request:
- use new Azure SDK
- drop support for azure credentials and azure profiles JSON files
  - the authentication is done using azidentity default credential stuff, more to read about it: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#section-readme
- drop obsolete commands in ore (for image replication, unreplication and sharing)
- deduplicate ore azure commands for image creation and blob uploading
- drop flags for azure auth (azure credentials location, azure profile location and use identity)
- drop the use of storage shared keys, storage accounts are created without the keys now

--

- [x] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)

jepio commented:
Github really fails to handle this many changes :/

I wanted to leave a comment: can you modify the storage account creation `AccountCreateParameters` to directly deploy `AllowSharedKeyAccess: false`.

krnowak commented:
> Github really fails to handle this many changes :/

I have split the work into separate commits, maybe will be a tad easier to review and for github to display stuff. The whole vendor stuff is now in a separate commit too. I also dropped fiddling with RBAC, so that shrunk the vendored code too (not pulling the msgraph stuff any more).

> 
> I wanted to leave a comment: can you modify the storage account creation `AccountCreateParameters` to directly deploy `AllowSharedKeyAccess: false`.

Done.

krnowak commented:
Tests:
- amd64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test/23977/
- arm64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test/23978/

Will need small changes in scripts on all branches (to drop obsolete `--azure-identity` flags).

jepio commented on platform/api/azure/storage.go:
```diff
@@ -16,98 +16,261 @@ package azure
-
-	return false, nil
+	// log-level=NONE only affects the log file - stdout is unaffected
+	cmd := exec.Command(azcopy, "cp", "--blob-type=PageBlob", "--log-level=NONE", sourceURL, dstSAS)
```
there is no lib for azcopy, and the the SDK CopyBlob function copies all 30GB of a sparse VHD instead of just the populated 1GB. But I think this code path is unused since we switched to the container pipeline (we don't copyblobs anymore, we always upload a vhd).

krnowak replied:
`azcopy` is doing similar thing to the `tryCopyPageBlob` function, but with goroutines and throttling and stuff. What I wasn't sure about is whether to keep the code that invokes `azcopy`.

krnowak commented:
> It's a bit dense to review but if the CI is green it's already a good sign.

Yeah, sorry about that. It's hard to split it into more commits, though.

> 
> > Will need small changes in scripts on all branches (to drop obsolete --azure-identity flags).
> 
> We don't need to forget about backporting this change to maintenance branches.

I have a PR (https://github.com/flatcar/scripts/pull/2063) for it. I think I'll just amend the PRs that will be created in scripts repo after merging this PR.

> 
> The dropped commands do not seem to be used in the Flatcar org.

Right, I was asking about them on matrix some time ago.

jepio commented:
can this be merged?
